### ISSUE TITLE
Add a property `headers` to TargetType?

### ISSUE THREAD

**Issue by EvilNOP** [Moya/Moya#1046]
I think it's better adding a `headers` property to TargetType protocol, it seems like the following:
```swift
var headers: [String : String]?
```

Say that there is an authentication token we need for case `isAuthenticationRequired`, we have to provide an endpoint when initialize `MoyaProvider`.

```swift
let endpointClosure = { (target: MyTarget) -> Endpoint<MyTarget> in
    let defaultEndpoint = MoyaProvider.defaultEndpointMapping(for: target)

    switch target {
    case .isAuthenticationRequired:
        return defaultEndpoint.adding(newHTTPHeaderFields: ["AUTHENTICATION_TOKEN": GlobalAppStorage.authToken])
    default:
        return defaultEndpoint.adding
    }
}

let provider = MoyaProvider<GitHub>(endpointClosure: endpointClosure)
```

Having the `headers` property set, I would't have to write the endpoint, I just need to write the code in MyTarget:
```swift
var headers: [String : String]? = {
    switch self {
    case .isAuthenticationRequired:
        return  ["AUTHENTICATION_TOKEN" : GlobalAppStorage.authToken]
    default:
        return nil
    }
}
```

**Comment by pedrovereza** [comment]
@EvilNOP Thanks for raising the idea 🙌  I've created #1067 with an initial implementation for `headers` inside `TargetType`. Feel free to review it and let us know what you think 😉

**Comment by pedrovereza** [comment]
Implemented in #1067
